keras_core/backend/jax/numpy.py

```diff
@@ -348,6 +353,7 @@ def less_equal(x1, x2):
 def linspace(
     start, stop, num=50, endpoint=True, retstep=False, dtype=None, axis=0
 ):
+    dtype = dtype or config.floatx()
```

Comment:
I think we should stick to the JAX dtype inference convention instead of NumPy, as it should be better suited for DL. What do you think?

We can consider reimplementing `jnp.result_dtype` for all backends
https://github.com/google/jax/blob/2cba122bbe512f7927d165fdbb29108dcf0fe124/jax/_src/dtypes.py#L638

It may require some time if we decide to do so.